cloudinit/config/schema.py

```diff
@@ -380,7 +391,7 @@ def _schemapath_for_cloudconfig(config, original_content):
             # Process non-list lines setting value if present
             list_index = 0
             key, value = line.split(":", 1)
-        if path_prefix:
+        if path_prefix and indent_depth > previous_depth:
```

Comment:
Yes this was the commit 3c8a84672f8c0d29ede65d1d9b3f16a25d1db2fe. 
Without this 'fix' the following cloud-config will Traceback because the `_schemapath_for_cloudconfig` would incorrectly represent the path key of runcmd as prefaced by YAML key

So this fails:
```bash

cat >test.yaml <<EOF
#cloud-config
apt_pipelining: bogus
runcmd: -1
EOF
 PYTHONPATH=. python3 -m cloudinit.cmd.main devel schema --annotate -c 1.yaml
....    errors_by_line[schemapaths[path]].append(msg)
KeyError: 'runcmd'
```

The KeyError is because `_schemapath_for_cloudconfig` was returning
{'apt_pipelining': 2, 'apt_pipelining.runcmd': 3}
instead of 
{'apt_pipelining': 2, 'runcmd': 3}


I added a unit test for this in the commit above.